packages/teuchos/comm/src/Teuchos_PerformanceMonitorBase.hpp

```diff
@@ -157,19 +154,22 @@ namespace Teuchos
   class PerformanceMonitorBase
   {
   public:
-    /** \brief Construct with a counter. */
+    //! Construct with a counter.
     PerformanceMonitorBase(T& counter_in, bool reset=false)
       : counter_(counter_in), isRecursiveCall_(counter_.isRunning())
     {
       (void) reset;  // get rid of "unused parameter" warning
       counter_.incrementNumCalls ();
     }
 
+    //! Default constructor is deleted, since it would be unsafe.
+    PerformanceMonitorBase () = delete;
+
     /// \brief Destructor.
     ///
     /// The destructor for the base class does nothing.  We provide a
     /// virtual destructor for memory safety of derived classes.
-    virtual ~PerformanceMonitorBase() {}
+    virtual ~PerformanceMonitorBase() = default;
```

Comment:
Why delete this?  What is the harm in having a virtual destructor with an empty implementation?

Reply:
`= default` declares intent.  I actually should have deleted the two lines of documentation above this, since "virtual default destructor" is a typical idiom for an abstract base class.

Using `= default` instead of `{}` matters more for non-virtual destructors, since it is necessary for the class to be [trivially destructible](https://en.cppreference.com/w/cpp/language/destructor#Trivial_destructor).

Reply:
Got it.  Thanks!